tests/multiprocessing_sync_tests.py

```diff
@@ -3,57 +3,77 @@
 
 from syllabus.tests import SyncTestCurriculum
 from syllabus.core import make_multiprocessing_curriculum, make_ray_curriculum
-from syllabus.tests import test_single_process, test_native_multiprocess, test_ray_multiprocess, create_synctest_env
+from syllabus.tests import create_synctest_env
+from syllabus.tests import test_single_process as helper_single_process
+from syllabus.tests import test_native_multiprocess as helper_native_multiprocess
+from syllabus.tests import test_ray_multiprocess as helper_ray_multiprocess 
 
+# SETUP Global Variables
 N_ENVS = 128
 N_EPISODES = 300
 
-if __name__ == "__main__":
+def evaluate_curriculum(curriculum, num_envs=N_ENVS, num_episodes=N_EPISODES):
+    stats = curriculum.get_stats()
+    expected_reward = 100 * num_envs * num_episodes
+    assert stats["total_reward"] == expected_reward, f"Curriculum total reward is {stats['total_reward']}, expected {expected_reward}"
+    for task, count in stats["task_counts"].items():
+        if task == 0:
+            assert count == 0, "Received completed error tasks, expected 0"
+        else:
+            assert count == num_envs, f"Curriculum task '{task}' count is {count}, expected {num_envs}"
+    expected_dones = num_envs * num_episodes
+    assert stats["total_dones"] == expected_dones, f"Curriculum total dones is {stats['total_dones']}, expected {expected_dones}"
+ 
+def generate_environment(num_episodes=N_EPISODES):
+    return create_synctest_env(env_args=(num_episodes,))
+
+def test_ray_initiation():
     ray.init()
```

Comment:
This doesn't actually test anything, but we need to call `ray.init` for the ray tests to work. Maybe just move `ray.init()` into the main script at the bottom instead of calling it in a test?

Reply:
Yeah, you are right. This just initializes ray environment which is required to test `ray_multiprocess`. Hence we could initialize it during testing of `ray_multiprocess` itself.